lib/SIL/IR/SILInstruction.cpp

```diff
@@ -1366,6 +1366,12 @@ bool SILInstruction::mayTrap() const {
   }
 }
 
+bool SILInstruction::maySynchronize() const {
+  // TODO: We need side-effect analysis and library annotation for this to be
+  //       a reasonable API.  For now, this is just a placeholder.
+  return false;
```

Comment:
Should maySynchronize return true for FullApplySite?

Reply:
Fixed.